bin/notify-users-on-new-locations

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""
+Notifies specified parties of new data for their interested locations
+"""
+import sys
+import json
+import argparse
+import subprocess
+import pandas as pd
+from pathlib import Path
+from typing import Dict, Iterable, List, Union
+
+
+LOCATION_HIERARCHY_COLUMNS = ['region', 'country', 'division', 'location']
+base = Path(__file__).resolve().parent.parent
+
+
+def create_new_areas_lists(metadata: pd.DataFrame) -> Dict[str, List]:
+    """
+    Returns a dictionary with keys of every location hierarchy resolution and
+    values containing a list of unique area names for each resolution in the
+    given *metadata*.
+    """
+    def unique_resolution_areas(series: Iterable[pd.Series]) -> List[str]:
+        """ Returns a list of unique, non-null items in the given *series*. """
+        return list(
+            pd.concat(series) \
+                .dropna() \
+                .unique()
+        )
+
+    # Locations do not have an associated exposure column
+    resolutions_with_exposure = LOCATION_HIERARCHY_COLUMNS.copy()
+    resolutions_with_exposure.remove('location')
+
+    areas: Dict[str, List] = {}
+
+    for r in resolutions_with_exposure:
+        areas[r] = unique_resolution_areas(
+            [incoming_metadata[r], incoming_metadata[f'{r}_exposure']]
+        )
+
+    # Handle location edge case
+    areas['location'] = unique_resolution_areas([incoming_metadata['location']])
+
+    return areas
+
+
+def alert_user_of_new_strains(entry: Dict[str, Union[str, List[str]]],
+    new_areas: Dict[str, List[str]]):
+    """
+    Using the given configuration *entry*, sends a Slack alert to the configured
+    user if there are new strains in the user's interested location hierarchy
+    areas.
+    """
+    member_id = entry.get('slack_member_id')
+
+    if not member_id:
+        # We can't send a Slack message without this field
+        return
+
+    alerts: List[str] = []
+
+    # Ignore the first two keys in the JSON which are the user's name and Slack
+    # member ID
+    for resolution in list(entry)[2:]:
```

Comment:
Perhaps it's better here to loop over the location hierarchy columns instead of the keys of the config object so I don't have this hardcoded assumption of `[2:]` here.